blockchain/chain_sync/src/sync.rs

```diff
@@ -134,12 +140,38 @@ impl<DB> ChainSyncer<DB>
 where
     DB: BlockStore,
 {
-    /// Starts syncing process
-    pub async fn sync(mut self) -> Result<(), Error> {
+    pub async fn start(mut self) -> Result<(), Error> {
         self.net_handler.spawn(Arc::clone(&self.peer_manager));
```

Comment:
I guess i didn't notice this last time, but why didnt you just do `self.peer_manager.clone()` seems to work fine when I try it...

Reply:
In this case it's the same, because the type is inferred, it's just recommended with Arc and Rc to explicitly clone like that so that you don't accidentally clone the underlying type and create a duplicate. Can switch if you'd like, but safer in future to keep imo

Reply:
That's cool. I didn't know this was convention. It's funny I think i saw someone proposing that they change the name of the clone method on Arc Rc the other day. Makes sense